pkg/mimir/sanity_check.go

```diff
@@ -66,7 +66,7 @@ func checkDirectoriesReadWriteAccess(
 	if cfg.isAnyModuleEnabled(All, Ruler, Backend) {
 		errs.Add(errors.Wrap(checkDirReadWriteAccess(cfg.Ruler.RulePath, dirExistFn, isDirReadWritableFn), "ruler"))
 	}
-	if cfg.isAnyModuleEnabled(AlertManager) {
+	if cfg.isAnyModuleEnabled(AlertManager, Backend) {
```

Comment:
Not strictly related to this PR, but I think that at this point we need a `cfg.isEnabledAnyModuleIncluding(AlertManager)` that should solve the dependency tree. Otherwise all these changes are very error-prone.

Reply:
Yes, I agree. I tried to it few weeks ago and was slightly harder than I though, because we have the dependency tree **after** we run most of the code currently calling `isAnyModuleEnabled()`. So, it needs some refactoring. I will look again at it and open an issue.

Reply:
https://github.com/grafana/mimir/issues/2844